code/modules/mob/dead/observer/observer.dm

```diff
@@ -312,8 +318,15 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	if(stat != DEAD)
 		succumb()
 	if(stat == DEAD)
-		ghostize(TRUE)
-		return TRUE
+		if(isethereal(usr))
+			var/response = tgui_alert(usr, "Are you sure you want to ghost? If you ghost whilst crystallizing you cannot re-enter your body!", "Confirm Ghost Observe", list("Ghost", "Stay in Body"))
+			if(response == "Ghost")
+				ghostize(FALSE)
+			else
+				return FALSE
+		else
+			ghostize(TRUE)
+			return TRUE
```

Comment:
check if they aren't an ethereal, and you can return the code early before doing all that tgui alert stuff. reads much nicer than making else statement pyramids